### PR TITLE
Automation for setting default dataset view

### PR DESCRIPTION
**Pull request by labkey-sweta**
#### Rationale
1. Adding test coverage for setting dataset default view. ( Issue 38340)
2. Adding test coverage for setting default view for dataset with name and label. (Issue 40502).

#### Related Pull Requests
https://github.com/LabKey/platform/pull/1222

#### Changes
1. Added some helper method for setting default view.
2. New page class for SetDefaultGridView.

### CODE REVIEW

File: src/org/labkey/test/components/SetDefaultPageClass.java [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+{
+    public SetDefaultPageClass(WebDriver driver)
+    {
+        super(driver);
+    }
+
+    public void selectDefaultGrid(String gridName)
+    {
+        click(Locator.linkWithText("select").withAttributeContaining("onclick",gridName));
+        clickAndWait(Locator.tagWithText("span","Done"));
```
Comment (cnathe) [inline src/org/labkey/test/components/SetDefaultPageClass.java]: Isn't the "Done" a button? I would have expected this to use one of the click button and wait helpers.

File: src/org/labkey/test/components/SetDefaultPageClass.java [hunk 2]
```diff
@@ -0,0 +1,19 @@
+package org.labkey.test.components;
+
+import org.labkey.test.Locator;
+import org.labkey.test.pages.LabKeyPage;
+import org.openqa.selenium.WebDriver;
+
+public class SetDefaultPageClass extends LabKeyPage
```
Comment (cnathe) [inline src/org/labkey/test/components/SetDefaultPageClass.java]: I believe this "Set Default" page only applies within the study module for datasets. Should this move down into the `test/components/study` dir?

File: src/org/labkey/test/util/DataRegionTable.java [hunk 3]
```diff
@@ -137,6 +138,15 @@ public PagingWidget getPagingWidget()
         return _pagingWidget;
     }
 
+    public SetDefaultPageClass clicksetDefault()
+    {
+        if (!getCustomizeView().isPanelExpanded())
```
Comment (cnathe) [inline src/org/labkey/test/util/DataRegionTable.java:144]: I'm curious why this if statement is needed. Does the "Set Default" option not work or show for some reason if the custom view grid is expanded?

**Review by cnathe — CHANGES_REQUESTED** [review]

File: src/org/labkey/test/util/DataRegionTable.java [hunk 4]
```diff
@@ -137,6 +138,15 @@ public PagingWidget getPagingWidget()
...
+        {
+            getViewsMenu().clickSubMenu(false, "Set Default");
```
Comment (labkey-tchad) [inline src/org/labkey/test/util/DataRegionTable.java]: This menu option navigates.
```suggestion
            getViewsMenu().clickSubMenu(true, "Set Default");
```

File: src/org/labkey/test/components/SetDefaultPageClass.java [hunk 5]
```diff
@@ -0,0 +1,19 @@
...
+{
+    public SetDefaultPageClass(WebDriver driver)
+    {
+        super(driver);
+    }
+
+    public void selectDefaultGrid(String gridName)
+    {
+        click(Locator.linkWithText("select").withAttributeContaining("onclick",gridName));
```
Comment (labkey-tchad) [inline src/org/labkey/test/components/SetDefaultPageClass.java]: This click triggers a page load:
```suggestion
        clickAndWait(Locator.linkWithText("select").withAttributeContaining("onclick",gridName));
```

File: src/org/labkey/test/components/SetDefaultPageClass.java (hunk 2 above)
Comment (labkey-tchad) [inline src/org/labkey/test/components/SetDefaultPageClass.java]: Page class names should correspond to the server action they wrap (ViewPreferencesAction) and the package name should correspond with the module containing that action (study).
```suggestion
package org.labkey.test.pages.study;

import org.labkey.test.Locator;
import org.labkey.test.pages.LabKeyPage;
import org.openqa.selenium.WebDriver;

public class ViewPreferencesPage extends LabKeyPage
```

**Review by labkey-tchad — CHANGES_REQUESTED** [review]

File: src/org/labkey/test/components/SetDefaultPageClass.java (hunk 2 above)
Comment (labkey-sweta) [reply src/org/labkey/test/components/SetDefaultPageClass.java]: Done

Comment (labkey-sweta) [reply src/org/labkey/test/components/SetDefaultPageClass.java]: ok

**Review by labkey-tchad — APPROVED** [review]

**Review by cnathe — APPROVED** [review]

**Pull request merged**